stablehlo_coreml/sort_utils.py

```diff
@@ -1,212 +1,227 @@
 import numpy as np
-from jaxlib.mlir import ir
 from jaxlib.mlir.dialects.stablehlo import (
-    CompareOp, SelectOp, ConstantOp
+    CompareOp, SelectOp, ConstantOp, OrOp, AndOp
 )
 from jax._src.lib.mlir.dialects import hlo
 
-
-def match_sort(comparator_root, args, inputs):
-    """
-    Analyzes the comparator region of a SortOp to determine if it implements a supported sorting pattern.
-    We try to analyze the comparator logic for multi-key sorts.
-
-    Multi-key sort compares multiple keys in sequence. If the primary keys are equal,
-    it moves to the secondary keys, and so on.
-
-    The comparator logic is expected to look like a chain of SelectOps:
-    if (k1 == k2) then compare(next_keys) else compare(k1 < k2)
-
-    Args:
-        comparator_root: The current operation in the comparator region being analyzed (starts at the return value).
-        args: The arguments of the comparator block (representing the two elements being compared).
-        inputs: The list of input tensors to the SortOp.
-
-    Returns:
-        A list of (tensor, ascending) tuples representing the sort keys, or None if the pattern doesn't match.
-    """
-    def get_op(val):
-        if isinstance(val.owner, ir.Block):
-            return None
-        return val.owner.opview
-
+from coremltools.converters.mil.mil import Builder as mb
+from coremltools.converters.mil.mil import types
+
+
+def bitcast_fp(x):
+    width = types.nptype_from_builtin(x.dtype)
+    ieee754 = {np.float32: (8, 23), np.float16: (5, 10)}
+    assert types.is_float(x.dtype) and width in ieee754
+    exponent, fraction = ieee754[width]
+    e_bias = 2 ** (exponent - 1) - 1
+    e_offset = 2 ** fraction
+
+    # Detect special values
+    # mb.is_nan/is_inf are not available, use comparisons
+    is_nan = mb.not_equal(x=x, y=x)
+    is_inf = mb.equal(x=mb.abs(x=x), y=width(float('inf')))
```

Comment:
nit: I think you can use `np.inf` instead of `float('inf')` which looks slightly cleaner